packages/esm-patient-search-app/src/utils/fhir-mapper.ts

```diff
@@ -71,3 +71,89 @@ export function mapToFhirPatient(patient: SearchedPatient) {
         })) ?? [],
   };
 }
+
+function calculateAgeFromBirthDate(birthDate?: string): number {
+  if (!birthDate) {
+    return 0;
+  }
+
+  try {
+    const dob = new Date(birthDate);
+
+    if (isNaN(dob.getTime())) {
+      return 0;
+    }
+
+    const today = new Date();
+    let age = today.getFullYear() - dob.getFullYear();
+    const monthDiff = today.getMonth() - dob.getMonth();
+
+    if (monthDiff < 0 || (monthDiff === 0 && today.getDate() < dob.getDate())) {
+      age--;
+    }
+
+    return Math.max(0, age);
+  } catch (error) {
+    return 0;
+  }
+}
+
+export function mapSearchedPatientFromFhir(patient: fhir.Patient): SearchedPatient {
+  const name = (patient.name && patient.name[0]) || ({} as fhir.HumanName);
+  const address = (patient.address && patient.address[0]) || ({} as fhir.Address);
+
+  const personNameDisplay = name?.text || [...(name?.given || []), name?.family].filter(Boolean).join(' ');
+
+  const mappedAddress: Address = {
+    preferred: true,
+    voided: false,
+    address1: address?.text || [address?.line?.[0], address?.line?.[1]].filter(Boolean).join(' '),
+    cityVillage: address?.city || '',
+    country: address?.country || '',
+    postalCode: address?.postalCode || '',
+    stateProvince: address?.state || '',
+  };
+
+  const mappedIdentifiers: Array<Identifier> = (patient.identifier || []).map((id) => ({
+    display: id.value || '',
+    identifier: id.value || '',
+    identifierType: {
+      uuid: (id.type?.coding && id.type.coding[0]?.code) || '',
+      display: id.type?.text || id.type?.coding?.[0]?.display || '',
+    } as any,
+    location: { uuid: '', display: '' } as any,
```

Comment:
Both of these should be typed better. `{ uuid: string; display: string }` should satisfy the contract.